src/elements/e-video.vue

```diff
@@ -0,0 +1,254 @@
+      },
+
+      /**
+       * Allows to set the video player height.
+       */
+      height: {
+        type: [String, Number],
+        default: 315,
+      },
+
+      /**
+       * Defines the players size to be responsive 100% width to the parent container
+       */
+      responsive: {
+        type: Boolean,
+        default: false,
+      },
+    },
+
+    // emits: {},
+    // setup(): Setup {},
+
+    computed: {
+      mapSrc(): string | undefined {
+        const playerBaseUrl = playerUrlSources[this.source];
+
+        if (!(this.videoId || this.videoUrl)) {
+          // eslint-disable-next-line no-console
+          console.error("Neither 'videoId' or 'videoUrl' were defined", this.$el);
+
+          return undefined;
+        }
+
+        if (this.videoUrl && !this.videoId) {
+          if (!this.validateUrl()) {
+            return undefined;
+          }
+
+          if (this.source === 'youtube') {
+            /**
+             * Extract id from video url
+             */
+            const id = this.videoUrl.substring(
+              this.videoUrl.indexOf('v=') + 2,
+              this.videoUrl.indexOf('&')
+            );
+
+            return `${playerBaseUrl}${id}`;
+          }
+
+          const id = this.videoUrl.substring(
+            this.videoUrl.indexOf(videoUrlSources[this.source])
+              + videoUrlSources[this.source].length,
+            this.videoUrl.length
+          );
+
+          return `${playerBaseUrl}${id}`;
+        }
+
+        if (this.videoId.includes(videoUrlSources[this.source])) {
+          // eslint-disable-next-line no-console
+          console.error('video-id can not be part of a URL');
+
+          return undefined;
+        }
+
+        return `${playerBaseUrl}${this.videoId}`;
+      },
+    },
+
+    // computed: {},
+    watch: {
+      /**
+       * This should check whether responsive or height + width are specified
+       */
+      $props: {
+        immediate: true,
+        handler() {
+          if (this.responsive && (this.width || this.height)) {
+            console.error('Either use responsive or height and width');
+          } else if (
+            (this.width && !this.height)
+            || (!this.width && this.height)
+          ) {
+            console.error('width and height both need to be specified');
+          }
+        },
+      },
+      videoUrl: {
+        immediate: true,
+        handler() {
+          this.validateUrl();
+        },
+      },
+    },
+
+    // beforeCreate() {},
+    // created() {},
+    // beforeMount() {},
+    // mounted() {},
+    // beforeUpdate() {},
+    // updated() {},
+    // activated() {},
+    // deactivated() {},
+    // beforeUnmount() {},
+    // unmounted() {},
+
+    methods: {
+      /**
+       * This is a helper function to validate the videoUrl property
+       */
+      validateUrl(): boolean {
+        if (
+          this.videoUrl
+          && !this.videoUrl.includes(videoUrlSources[this.source])
```

Comment:
todo: please use `startsWith()` instead. Else urls like: `https://www.evildomain.com?foo=https://youtube.com` could be used.